tests/integration/common/transaction_utils.rs

```diff
@@ -79,40 +81,155 @@ pub fn to_internal_tx(account_tx: &AccountTransaction) -> InternalTransaction {
     let contract_address_salt: Option<Felt252> = None;
     let contract_hash: Option<Felt252> = None;
     let constructor_calldata: Option<Vec<Felt252>> = None;
-    let nonce: Option<Felt252> = None;
+    let nonce: Option<Felt252>;
     let sender_address: Option<Felt252>;
     let entry_point_selector: Option<Felt252>;
     let entry_point_type: Option<String> = Some("EXTERNAL".to_string());
     let signature: Option<Vec<Felt252>>;
-    let class_hash: Option<Felt252> = None;
+    let class_hash: Option<Felt252>;
     let compiled_class_hash: Option<Felt252> = None;
     let calldata: Option<Vec<Felt252>>;
     let paid_on_l1: Option<bool> = None;
-    let r#type: String = "INVOKE_FUNCTION".to_string();
+    let r#type: String;
     let max_fee: Option<Felt252>;
 
     match account_tx {
-        Declare(_) => panic!("Not implemented"),
```

Comment:
The changes here to `fn to_internal_tx` are pretty messy, I think this will need a refactor in order to handle all txn types we anticipate. Duplicating it for each high-level txn type is probably ok (at the expense of some code duplication).

Reply:
See https://github.com/keep-starknet-strange/snos/pull/177#discussion_r1627347326